OutOfSchool/OutOfSchool.WebApi/Services/Images/ImageService.cs

```diff
@@ -0,0 +1,263 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Runtime.InteropServices.ComTypes;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OutOfSchool.Services.Common.Exceptions;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Models.Images;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.WebApi.Common;
+using OutOfSchool.WebApi.Common.Resources;
+using OutOfSchool.WebApi.Common.Resources.Codes;
+using OutOfSchool.WebApi.Config.Images;
+using OutOfSchool.WebApi.Extensions;
+using OutOfSchool.WebApi.Models;
+
+namespace OutOfSchool.WebApi.Services.Images
+{
+    /// <summary>
+    /// Provides APIs for operations with images.
+    /// </summary>
+    public class ImageService : IImageService
+    {
+        private readonly IExternalImageStorage externalStorage;
+        private readonly IImageRepository imageRepository;
+        private readonly IServiceProvider serviceProvider;
+        private readonly ILogger<ImageService> logger;
+
+        public ImageService(IExternalImageStorage externalStorage, IWorkshopRepository workshopRepository, IImageRepository imageRepository, IServiceProvider serviceProvider, ILogger<ImageService> logger)
+        {
+            this.externalStorage = externalStorage;
+            this.imageRepository = imageRepository;
+            this.serviceProvider = serviceProvider;
+            this.logger = logger;
+        }
+
+        /// <inheritdoc/>
+        public async Task<Result<ImageStorageModel>> GetByIdAsync(Guid imageId)
+        {
+            logger.LogInformation($"Getting image by id = {imageId} was started.");
+            var imageMetadata = await imageRepository.GetMetadataById(imageId).ConfigureAwait(false);
```

Comment:
probably leave a comment here too if we stick with Mongo.
Two step image retrieval (get info from sql and then image from mongo) will be a very resource intensive operation. As this method is used in public endpoint - we can use just image id and go to mongo directly.